users/views.py

```diff
@@ -1,9 +1,18 @@
 from rest_framework import generics
 from rest_framework.permissions import AllowAny
 
-from users.serializers import SignupSerializer
+from instagram_own_version.permissions import IsOwner
+from users.models import User
+from users.serializers import SignupSerializer, UserPrivateSerializer
 
 
 class SignupView(generics.CreateAPIView):
     permission_classes = (AllowAny,)
     serializer_class = SignupSerializer
+
+
+class UpdateProfileView(generics.RetrieveUpdateAPIView):
+    queryset = User.objects.all()
+    permission_classes = (IsOwner,)
+    serializer_class = UserPrivateSerializer
+    lookup_field = "id"
```

Comment:
Please remove this line as PK (primary key which is ID in our case) is the lookup field by default. Just update `id` to `pk` in the url

Reply:
Also please add something like this to manage data returned by the endpoint: 
```
    def get_serializer_class(self):
        # todo cache object to not query the database twice
        if self.request.user == self.get_object():
            return UserPrivateSerializer
        return UserPublicSerializer
```

Reply:
Done